src/main/java/uk/gov/justice/digital/delius/service/CustodyService.java

```diff
@@ -8,44 +8,55 @@
 import uk.gov.justice.digital.delius.data.api.Custody;
 import uk.gov.justice.digital.delius.data.api.UpdateCustody;
+import uk.gov.justice.digital.delius.jpa.standard.entity.CustodyHistory;
 import uk.gov.justice.digital.delius.jpa.standard.entity.Event;
+import uk.gov.justice.digital.delius.jpa.standard.entity.Offender;
 import uk.gov.justice.digital.delius.jpa.standard.entity.RInstitution;
+import uk.gov.justice.digital.delius.jpa.standard.repository.CustodyHistoryRepository;
 import uk.gov.justice.digital.delius.jpa.standard.repository.InstitutionRepository;
+import uk.gov.justice.digital.delius.jpa.standard.repository.OffenderRepository;
 import uk.gov.justice.digital.delius.transformers.ConvictionTransformer;
 
+import java.time.LocalDate;
 import java.util.Map;
 
 @Service
 @Slf4j
 public class CustodyService {
     private final Boolean updateCustodyFeatureSwitch;
     private final TelemetryClient telemetryClient;
-    private final OffenderService offenderService;
+    private final OffenderRepository offenderRepository;
     private final ConvictionService convictionService;
     private final InstitutionRepository institutionRepository;
     private final ConvictionTransformer convictionTransformer;
+    private final CustodyHistoryRepository custodyHistoryRepository;
+    private final ReferenceDataService referenceDataService;
 
     public CustodyService(
             @Value("${features.noms.update.custody}")
-            Boolean updateCustodyFeatureSwitch,
+                    Boolean updateCustodyFeatureSwitch,
             TelemetryClient telemetryClient,
-            OffenderService offenderService,
+            OffenderRepository offenderRepository,
             ConvictionService convictionService,
             InstitutionRepository institutionRepository,
-            ConvictionTransformer convictionTransformer) {
+            ConvictionTransformer convictionTransformer,
+            CustodyHistoryRepository custodyHistoryRepository,
+            ReferenceDataService referenceDataService) {
         this.updateCustodyFeatureSwitch = updateCustodyFeatureSwitch;
         this.telemetryClient = telemetryClient;
-        this.offenderService = offenderService;
+        this.offenderRepository = offenderRepository;
         this.convictionService = convictionService;
         this.institutionRepository = institutionRepository;
         this.convictionTransformer = convictionTransformer;
+        this.custodyHistoryRepository = custodyHistoryRepository;
+        this.referenceDataService = referenceDataService;
     }
 
     @Transactional
     public Custody updateCustody(final String nomsNumber,
                                  final String bookingNumber,
                                  final UpdateCustody updateCustody) {
-        final var maybeOffender = offenderService.getOffenderByNomsNumber(nomsNumber);
+        final var maybeOffender = offenderRepository.findByNomsNumber(nomsNumber);
```

Comment:
switched to use raw entity since we need it later for the custody history entity